pkg/sources/chunker.go

```diff
@@ -26,20 +26,24 @@ func Chunker(originalChunk *Chunk) chan *Chunk {
 		r := bytes.NewReader(originalChunk.Data)
 		reader := bufio.NewReaderSize(bufio.NewReader(r), ChunkSize)
 		for {
-			chunkBytes := make([]byte, ChunkSize)
+			chunkBytes := make([]byte, ChunkSize+PeekSize)
 			chunk := *originalChunk
+			chunkBytes = chunkBytes[:ChunkSize]
 			n, err := reader.Read(chunkBytes)
 			if err != nil && !errors.Is(err, io.EOF) {
 				break
 			}
-			peekData, _ := reader.Peek(PeekSize)
-			chunk.Data = append(chunkBytes[:n], peekData...)
-			if n > 0 {
-				chunkChan <- &chunk
-			}
-			if errors.Is(err, io.EOF) {
+			if n == 0 {
+				if errors.Is(err, io.EOF) {
```

Comment:
This is going to `break` either way, so the extra check for `EOF` isn't necessary.

Reply:
yea my bad, switched it to a continue for non-eof

Reply:
I think we only want to break for EOF and continue for non-EOF, is that correct? or do we break for both cases.